mfr/extensions/tabular/libs/h5py_scipy_tools.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict
+
+import h5py
+import scipy.io
+
+from mfr.extensions.tabular import exceptions as tabular_exceptions
+from mfr.extensions.tabular.utilities import data_population, header_population
+
+
+def build_sheets(name, list_data, sheets):
+    # add default header to fix rendering of some data formats
+    header = [i for i in range(1, 1 + len(list_data[0]))]
+    values = data_population(list_data, headers=header)
+    header = header_population(header)
+    sheets[str(name)] = (header, values)
+    return sheets
+
+
+def mat_v73(fp):
+    """Read and convert a mat v7.3+ file to JSON format using h5py
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    # workbook cannot be empty or none. Will atleast have some sort of data in it
+    workbook = h5py.File(fp.name, 'r')
+    sheets = OrderedDict()
+    variables = workbook.items()
+
+    for var in variables:
+        name = var[0]
+        data = var[1]
+        if type(data) is h5py.Dataset:
+            # h5py Uses row-major ordering. this fixes it so it displays like it does in matlab
+            # basically just flip it on its axis
+            list_data = list(zip(*data.value.tolist()))
+            sheets = build_sheets(name, list_data, sheets)
+
+    return sheets
+
+
+def mat_v7(fp):
+    """Read and convert a mat v7.0 and below file to JSON format using scipy.io.matload
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+    try:
+        # workbook cannot be empty or none, will always have header information
+        workbook = scipy.io.loadmat(fp.name)
+    except NotImplementedError as e:
+        raise e
+    except Exception:
+        raise(tabular_exceptions.UnexpectedFormattingError(
+            'Cannot render this file at this time.'
+            ' The file may not be a .mat file, or may be corrupted'))
+
+    sheets = OrderedDict()
+
+    for key in ('__header__', '__version__', '__globals__'):
+        workbook.pop(key, None)
+
+    for key in workbook:
+        list_data = workbook[key].tolist()
+        sheets = build_sheets(key, list_data, sheets)
+
+    return sheets
+
+def mat_h5py_scipy(fp):
+    try:
```

Comment:
Please add a blurb explaining what you're doing (trying to load a file w/ two different libraries), why you're doing it (h5py doesn't support <v7.3, scipy doesn't support >v7.0), and why you try one before the other (even if it's arbitrary).

In a related note, am I misunderstanding the naming or does this not support mat files v7.1 and v7.2?  If so, that's weird enough that it might be worth a bespoke error message.

Reply:
http://www.mathworks.com/help/matlab/import_export/mat-file-versions.html 
From my understanding there is no 7.1 or 7.2 (there may be a 7.1, but from what I've read scipy can open it if it exists).

The file types use different compressions and can store differing amounts of data. It seems they are just named after whatever matlab version they are released with (? just a guess).